### PR TITLE
 Account for query bit when switching on txn version for hashing

### PR DESCRIPTION
**Pull request by omerfirmak**

### CODE REVIEW

File: core/transaction.go [hunk 1]
```diff
@@ -70,20 +72,65 @@ type Transaction interface {
...
 
 const (
 	// Calculated at https://hur.st/bloomfilter/?n=1000&p=&m=8192&k=
 	// provides 1 in 51 possibility of false positives for approximately 1000 elements
 	eventsBloomLength    = 8192
 	eventsBloomHashFuncs = 6
+	queryBit             = 128
 )
 
+type TransactionVersion felt.Felt
```
Comment (joshklop) [inline core/transaction.go:91]: Should we add a note here that we need a migration if we change the underlying type of `TransactionVersion`?

Comment (omerfirmak) [reply core/transaction.go:91]: Will do but I think one should already be aware that all `core` types are used as storage objects and be careful when making any changes on them.

File: vm/transaction.go [hunk 2]
```diff
@@ -23,7 +18,8 @@ func marshalTxn(txn core.Transaction) (json.RawMessage, error) {
 		return nil, err
 	}
 
-	t.Version = clearQueryBit(t.Version)
+	versionWithoutQueryBit := (*core.TransactionVersion)(t.Version).WithoutQueryBit()
+	t.Version = versionWithoutQueryBit.AsFelt()
```
Comment (joshklop) [inline vm/transaction.go:22]: Can we do `txn.WithoutQueryBit` before we convert to the feeder type?

**Review by joshklop — APPROVED** [review]

File: vm/transaction.go (hunk 2 above)
Comment (omerfirmak) [reply vm/transaction.go:22]: Sadly no, we dont have access to `Version` field thru `core.Transaction` interface.
Alternative could be doing it in the switch below, but it is far from DRY.

**Pull request merged**